### DISCUSSION TITLE
Error using linprog

### DISCUSSION THREAD

**Discussion by pierre321** [yalmip/YALMIP#1318] [Problems using YALMIP] [answered]
I installed a new Matlab Version (R 2023a) and wanted to try out Yalmip which I have used successfully for many years on Matlab R2022a.
In order to test it, I used a small linear algorithm adopted from the getting stared example.

```
yalmip('clear')

% Define variables
x = sdpvar(2,1);

% Define constraints 
Constraints = [sum(x) <= 4, 0 <= x(1) <= 3, 0.5 <= x(2) <= 1.5];

% Define an objective
Objective = -(x(1)+x(2));

% Set some options for YALMIP and solver
options = sdpsettings('verbose',1,'solver','quadprog','quadprog.maxiter',100);

% Solve the problem
sol = optimize(Constraints,Objective,options);

% Analyze error flags
if sol.problem == 0
 % Extract and display value
 solution = value(x)
else
 display('Hmm, something went wrong!');
 sol.info
 yalmiperror(sol.problem)
end
```

--------------------------------------------------------------------------------------------------------------------------------------------

While the getting started example works just fine, I'm getting a weird warning for this example:

> Hmm, something went wrong!
> 
> ans =
> 
>     'Unknown problem in solver (Turn on 'debug' in sdpsettings) (Error using linprog
>      LINPROG(f,A,b,Aeq,beq,LB,UB,X0,OPTIONS) does not accept X0. Use LINPROG(f,A,b,Aeq,beq,LB,UB,OPTIONS) instead.)'
> 
> 
> ans =
> 
>     'Unknown problem in solver (Turn on 'debug' in sdpsettings) '
> 

------------------------------------------------------------------------------------------------------------------------------------------

If I turn on debug in the otpions, I'm getting the following error:

> Error using linprog
> LINPROG(f,A,b,Aeq,beq,LB,UB,X0,OPTIONS) does not accept X0. Use LINPROG(f,A,b,Aeq,beq,LB,UB,OPTIONS) instead.
> 
> Error in callquadprog>callsolver (line 56)
>         [x,fmin,flag,output,lambda] = linprog(model.c, model.A, model.b, model.Aeq, model.beq, model.lb, model.ub, model.x0,model.ops);
> 
> Error in callquadprog (line 12)
> solveroutput = callsolver(model,options);
> 
> Error in solvesdp (line 368)
>     eval(['output = ' solver.call '(interfacedata);']);
> 
> Error in optimize (line 31)
> [varargout{1:nargout}] = solvesdp(varargin{:});
> 
> Error in test1 (line 18)
> sol = optimize(Constraints,Objective,options);

 
What have I done wrong?

**Comment by johanlofberg** [comment, marked as answer]
It looks like mathworks have introduced a breaking change in their linprog solver in R2023. To get it to work you ill have to edit calllinprog and remove x0 from the argument list as indicated by the error message

**Reply by johanlofberg** [reply]
unless you have an inconsistent linprog file from mosek or gurobi or sometthing else in your path

which linprog -all
C:\Program Files\Matlab\R2022a\toolbox\optim\optim\linprog.m

**Reply by pierre321** [reply]
Thank you for your quick answer. I did as you told but now I'm getting a different error. The file is attached ([linprog.txt](https://github.com/yalmip/YALMIP/files/12081385/linprog.txt)).

> Error using linprog

> Too many input arguments.
> 
> Error in callquadprog>callsolver (line 56)
>         [x,fmin,flag,output,lambda] = linprog(model.c, model.A, model.b, model.Aeq, model.beq, model.lb, model.ub, model.x0,model.ops);
> 
> Error in callquadprog (line 12)
> solveroutput = callsolver(model,options);
> 
> Error in solvesdp (line 368)
>     eval(['output = ' solver.call '(interfacedata);']);
> 
> Error in optimize (line 31)
> [varargout{1:nargout}] = solvesdp(varargin{:});
> 
> Error in test1 (line 18)
> sol = optimize(Constraints,Objective,options);

**Reply by pierre321** [reply]
Would you recommand using another solver like gurobi or mosek instead?

**Reply by johanlofberg** [reply]
Gurobi/mosek/xpress is almost always preferred over linprog

But from the display above, you have not deleted the x0 argument as it rather clearly is still there in the second to last place.

**Reply by pierre321** [reply]
I deleted it as an input argument for the function: 

`function [x,fval,exitflag,output,lambda]=linprog(f,A,B,Aeq,Beq,lb,ub,options)`

It still appears several time within the function and I don't really have the knowledge to make changes in the function.

I think it's easier if I just use gurobi or stay with Matlab22. Again, thank you very much for your answers.

**Reply by johanlofberg** [reply]
what file are you editing? The crashing line to edit is is callquadprog>callsolver (line 56) as it still according to your error message calls linprog with too many arguments

**Reply by pierre321** [reply]
The linprog.m file

**Reply by johanlofberg** [reply]
You should not touch linprog files. You only edit YALMIPs callquadprog

**Reply by pierre321** [reply]
Sorry, this was my error. I applied the changes to callquadprod and it works now!!!

**Comment by pierre-haessig** [comment]
Hello,
Should an issue be opened for this? Issue https://github.com/yalmip/YALMIP/issues/1320, which I believe is the exact same problem, was opened but then converted to discussion. Is it better to reopen a new one?

**Comment by pierre-haessig** [comment]
For the reference, the `x0` parameter for warm-starting Matlab `linprog` isn't listed in the doc of the oldest release I could find ([Matlab R2018b linprog](https://fr.mathworks.com/help/releases/R2018b/optim/ug/linprog.html)). Still, it seems it was accepted until R2022b or so.
